brave/brave-core
# Fixes estimated earnings and ad notifications are not updating for users

Resolves https://github.com/brave/brave-browser/issues/11952

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

- Confirm catalog is downloaded upon browser launch
- Confirm catalog is downloaded every ~2 hours
- Confirm catalog download is not started if currently retrying a previously failed download
- Confirm unblinded payment tokens are redeemed ~24 hours
- Confirm unblinded payment tokens redemption is not started if currently retrying a previously failed download
- Confirm unblinded tokens are refilled if we have less than 10 unblinded tokens when triggering an event to server an ad notification
- Confirm failed confirmations are retried after attempting to download the catalog
- Confirm failed confirmations are retried when an unblinded token redemption fails
- Confirm unblinded tokens are refilled to 50 if we are below the threshold after attempting to download the catalog
- Confirm unblinded tokens are refilled to 50 if we are below the threshold after redeeming an unblinded token
- Confirm ad rewards are reconciled upon browser launch
- Confirm ad rewards are reconciled after restoring a wallet
- Confirm ad rewards are reconciled after redeeming unblinded payment tokens
- Confirm ad rewards are reconciled after claiming an ad promotion
- Confirm unblinded payment tokens redemption is not started if currently retrying a previously failed download

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
